modules/cudn-status-conditions.adoc

```diff
@@ -80,3 +80,157 @@ h|Message
+3+|One of the following messages is returned when the `spec.network.localnet.role` is set incorrect:
+h|Reason
+h|Message
+h|Resolution
+
+|The `role` field must be set for your localnet topology.
+|`spec.network.localnet.role: Required value`
+|You must set the `role` field.
+
+|`Primary` is not a supported value for the Localnet topology.
+|`spec.network.localnet.role: Unsupported value: "Primary": supported values: "Secondary"`
+|You must set the `role` field for your Localnet topology to `Secondary`-the accepted value.
+|===
+
+.Invalid `subnets` and `ipam` scenarios types for the `ClusterUserDefinedNetwork` CR
+[cols="2a,2a,3a,3a",options="header"]
+|===
+
+|Condition type
+3+|Reason, Message, Resolution
+
+.9+|`LocalnetInvalidSubnets`
+3+|One of the following messages is returned when either the `spec.network.localnet.subnets` or `spec.network.localnet.ipam` is set incorrect:
+h|Reason
+h|Message
+h|Resolution
+
+|The optional fields, `subnets` and `ipam.mode`, have to be set together.
+|`Subnets is required with ipam.mode is Enabled or unset, and forbidden otherwise`
+|You must set the `subnets` field unless the `spec.network.localnet.ipam.mode` is explicitly disabled.
+
+|The `spec.network.localnet.subnets` must have an acceptable value when using this optional field.
+|`The ClusterUserDefinedNetwork "localnet-empty-subnets-fail" is invalid: spec.network.localnet.subnets: Invalid value: 0: spec.network.localnet.subnets in body should have at least 1 items`
+|You must set an acceptable value for `spec.network.localnet.subnets`. Acceptable values are IPv4 and IPv6 Classless Inter-Domain Routing (CIDR) ranges that do not overlap with any CIDR ranges used by {product-title}.
+
+|The `subnet` field must be set when using the optional `spec.network.localnet.excludeSubnets` field.
+|`excludeSubnets must be unset when subnets is unset`
+|You must set the `spec.network.localnet.subnets` field when using the `spec.network.localnet.excludeSubnet` field.
+
+|The CIDR range is invalid.
+|`The ClusterUserDefinedNetwork "localnet-subnets-invalid-ipv4-cidr-fail" is invalid: spec.network.localnet.subnets[0]: Invalid value: "string": CIDR is invalid`
+|You must set an acceptable CIDR range for `spec.network.localnet.subnets` field. Acceptable values are IPv4 and IPv6 CIDR ranges which are not in use or reserved by {product-title}.
+
+|You must set the `subnets` field when the `ipam.mode` is `Enabled` or when the IPAM mode is unset because the default value is `Enabled`.
+|`Subnets is required with ipam.mode is Enabled or unset, and forbidden otherwise`.
+|You must set the `spec.network.localnet.subnets` field unless the `spec.network.localnet.ipam.mode` is explicitly disabled.
+
+|Setting two CIDR ranges for `spec.network.localnet.subnets` field requires that one be IPv4 and the other be IPv6.
+|`Invalid value...When 2 CIDRs are set, they must be from different IP families`.
+|You must change one of your CIDR ranges to a different IP family.
+
```

Comment:
@JoeAldinger Need to another error message
The ClusterUserDefinedNetwork "sec-localnet-net" is invalid: spec.network.localnet.excludeSubnets: Invalid value: "object": excludeSubnets must be subnetworks of the networks specified in the subnets field

This happens when excludeSubnet has CIDR that does not lie within the subnet CIDR
<pre>
apiVersion: k8s.ovn.org/v1
kind: ClusterUserDefinedNetwork
metadata:
  name: sec-localnet-net
spec:
  namespaceSelector:
    matchExpressions:
    - key: kubernetes.io/metadata.name
      operator: In
      values: ["a1", "a2"]
  network:
    topology: Localnet
    localnet:
      role: Secondary
      physicalNetworkName: mylocalnet
      subnets: ["192.168.100.0/24"]
      excludeSubnets: ["192.168.200.1/32"]
</pre>

Reply:
probably this error should also be described in the docs ...

Reply:
I would be OK with that. We may want to include the information on limit on excludeSubnet items also which when hit generates an error message *** spec.network.localnet.excludeSubnets: Too many: 26: must have at most 25 items**